File: Fundamentals of Discrete Mathematics/Combinаtorics/C++/task_2.cpp
#include <iostream>
using namespace std;

int main() {
    int count = 0;
    for (int i = 0; i < 1000; i++) {
        int num = i;
        int digit_count = 0;
        while (num > 0) {
            if (num % 10 == 7) {
                digit_count++;
            }
            num /= 10;
        }
        if (digit_count == 1) {
            count++;
        }
    }
    cout << "Number of integers between 0 and 999 containing exactly one digit 7: " << count << endl;
    return 0;
}

File: Fundamentals of Discrete Mathematics/Combinаtorics/Java/task_2.java
public class task_2 {
    public static void main(String[] args) {
        int count = 0;
        for (int i = 0; i < 1000; i++) {
            if (countDigit(i, 7) == 1) {
                count++;
            }
        }
        System.out.println("Number of integers between 0 and 999 containing exactly one digit 7: " + count);
    }

    private static int countDigit(int n, int digit) {
        int count = 0;
        while (n > 0) {
            if (n % 10 == digit) {
                count++;
            }
            n /= 10;
        }
        return count;
    }
}

